plugins/module_utils/system_access_users_utils.py

```diff
@@ -0,0 +1,855 @@
+
+        Note:
+            This property should be consulted before performing a save operation to avoid
+            unnecessary writes to the system configuration when no changes have been made.
+        """
+
+        return self._load_users() != self._users or self._load_groups() != self._groups
+
+    def _update_user_groups(self, user: User, existing_user: Optional[User] = None):
+        """
+        Manages the association of a user with specified groups, either by updating the groups of an
+        existing user or adding a new user to the appropriate groups. This method ensures that the
+        user is a member of all specified groups, adding the user to any groups they are not already
+        a part of, and maintains the integrity of group memberships across updates.
+        If user.groupname is None, the user is removed from all groups.
+
+        Parameters:
+            user (User): The user whose group memberships are to be updated. This includes both new
+                        users and users whose group memberships might change.
+            existing_user (Optional[User]): If the user already exists, this parameter should be the
+                                            user's current information. It is used to determine if
+                                            the existing group memberships need to be updated.
+
+        Raises:
+            OPNSenseGroupNotFoundError: If a specified group does not exist on the instance, this
+                                        exception is raised, indicating the need for corrective
+                                        action or error handling.
+        """
+        target_user = existing_user if existing_user else user
+
+        if user.groupname is None or not hasattr(user, "groupname"):
+            for existing_group in self._groups:
+                if existing_group.check_if_user_in_group(target_user):
+                    existing_group.remove_user(target_user)
+            return  # Exit the method after removing the user from all groups.
+
+        # Convert groupname to a list if it's not already.
+        group_names = user.groupname if isinstance(user.groupname, list) else [user.groupname]
+
+        for group_name in group_names:
+            group_found = False
+            for index, existing_group in enumerate(self._groups):
+                if existing_group.name == group_name:
+                    group_found = True
+                    if not existing_group.check_if_user_in_group(target_user):
+                        existing_group.add_user(target_user)
+                        self._groups[index] = existing_group
+                    break  # Stop searching once the group is found
+
+            if not group_found:
+                # Group was not found, raise an exception
+                raise OPNSenseGroupNotFoundError(f"Group '{group_name}' not found on Instance")
```

Comment:
This exception is not excepted anywhere, which will cause the module to send the traceback to the user. Since this is a user error I would make use of `module.fail_json` and present the exceptions' msg to the user. Same story for the other custom exception types caused by user error such as `OPNSenseNotValidBase32APIKeyError`

Reply:
Thank you @rekup! Fixed in f14438a.